src/assert.ts

```diff
@@ -29,7 +29,8 @@ export class AssertionError extends Error {
 
 /** https://docs.tsafe.dev/assert */
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-export function assert<_T extends true>(condition: any = true, msg?: string): asserts condition {
+export function assert<_T extends true>(condition: any, msg?: string): asserts condition {
+    if (arguments.length === 0) { condition = true; } // hack for backwards compat, see issue #6
```

Comment:
I don't know if this hack is even a good idea, just throwing it out there. The real fix is making `condition` not default to true.

Reply:
Yes this is what have to be done.  
We need the condition to be default to true for `assert<Equals<A,B>>()`.